ssvsigner/tls/tls.go

```diff
@@ -0,0 +1,495 @@
+
+	// Case 3: Only known clients file provided without server certificate
+	// This is invalid because a server must have a certificate to accept TLS connections
+	if c.ServerKeystoreFile == "" && c.ServerKnownClientsFile != "" {
+		return fmt.Errorf("server keystore file is required when specifying known clients file")
+	}
+
+	// Case 4: Server keystore and password file provided (with or without known clients)
+	// This is a valid configuration
+	return nil
+}
+
+// validateClientTLS validates the client TLS configuration.
+// It verifies that TLS configuration is consistent based on multiple valid use cases.
+func (c *Config) validateClientTLS() error {
+	// Case 1: No client TLS config provided - TLS is optional
+	if c.ClientKeystoreFile == "" && c.ClientServerCertFile == "" {
+		return nil
+	}
+
+	// Case 2: Only trusted server certificate file provided without client certificate
+	// This is valid - the client can verify server certificates without presenting its own
+	if c.ClientKeystoreFile == "" && c.ClientServerCertFile != "" {
+		return nil
+	}
+
+	// Case 3: Client keystore file provided but no password file
+	if c.ClientKeystoreFile != "" && c.ClientKeystorePasswordFile == "" {
+		return fmt.Errorf("client keystore password file is required when using a client keystore file")
+	}
+
+	// Case 4: Client keystore and password file provided (with or without server certificate)
+	// This is a valid configuration for mutual TLS
+	return nil
+}
+
+// loadClientCertificate loads the client certificate and private key from keystore.
+// It handles reading the password file and loading the certificate from the PKCS12 keystore.
+func (c *Config) loadClientCertificate() (tls.Certificate, error) {
+	password, err := loadPasswordFromFile(c.ClientKeystorePasswordFile)
+	if err != nil {
+		return tls.Certificate{}, fmt.Errorf("load client keystore password: %w", err)
+	}
+
+	return loadKeystoreCertificate(c.ClientKeystoreFile, password)
+}
+
+// loadServerCertificate loads the server certificate and private key from keystore.
+// It handles reading the password file and loading the certificate from the PKCS12 keystore.
+func (c *Config) loadServerCertificate() (tls.Certificate, error) {
+	password, err := loadPasswordFromFile(c.ServerKeystorePasswordFile)
+	if err != nil {
+		return tls.Certificate{}, fmt.Errorf("load server keystore password: %w", err)
+	}
+
+	return loadKeystoreCertificate(c.ServerKeystoreFile, password)
+}
+
+// loadServerFingerprints loads the trusted server fingerprints from a PEM certificate file.
+// It extracts the certificate's fingerprint and identity (common name or DNS name).
+func (c *Config) loadServerFingerprints() (map[string]string, error) {
+	serverCert, err := loadPEMCertificate(c.ClientServerCertFile)
+	if err != nil {
+		return nil, fmt.Errorf("load trusted server certificate: %w", err)
+	}
+
+	fingerprint := sha256.Sum256(serverCert.Raw)
+	fingerprintHex := strings.ToLower(hex.EncodeToString(fingerprint[:]))
```

Comment:
it's always lowercase
```suggestion
	fingerprintHex := hex.EncodeToString(fingerprint[:])
```